### ISSUE TITLE
UnifiedDiffParserException on apparently sane file (2nd variant)

### ISSUE THREAD

**Issue by moio** [java-diff-utils/java-diff-utils#107]
**Describe the bug**
`UnifiedDiffReader.parseUnifiedDiff` throws UnifiedDiffParserException on a patch which apparently looks good: 

https://raw.githubusercontent.com/bazelbuild/bazel/master/third_party/aws-sdk-auth-lite/patches-vs-1.11.297/01-bazel-strip-unused.patch

**To Reproduce**
Run this program:

```java
String patch = new Scanner(new URL(
    "https://raw.githubusercontent.com/bazelbuild/bazel/master/third_party/aws-sdk-auth-lite/patches-vs-1.11.297/01-bazel-strip-unused.patch"
).openStream(), "UTF-8").useDelimiter("\\A").next();

UnifiedDiffReader.parseUnifiedDiff(new StringBufferInputStream(patch));
```

**Expected behavior**

No Exceptions.

**Actual behavior**

Dec 14, 2020 9:49:46 PM com.github.difflib.unifieddiff.UnifiedDiffReader processLine
WARNING:   >>> no rule matched \ No newline at end of file
Exception in thread "main" com.github.difflib.unifieddiff.UnifiedDiffParserException: expected file start line not found
	at com.github.difflib.unifieddiff.UnifiedDiffReader.parse(UnifiedDiffReader.java:95)
	at com.github.difflib.unifieddiff.UnifiedDiffReader.parseUnifiedDiff(UnifiedDiffReader.java:146)
...

**System**
 - Java version: openjdk-11
 - Version: 4.9

**Notes**

:wave: java-diff-utils maintainers again! Here is another corner case I found. The patch comes from the Bazel project, so once again it's OK to include in tests.

**Comment by wumpz** [comment]
That is a nice huge diff. The new implementation of java-diff-utils will gain a lot of testing on this large file base. I will look into it.

**Comment by moio** [comment]
Thanks a lot @wumpz! I might have other cases as well for you later, as I plan to test each and every patch from CentOS and likely other open source projects as well :wink:

**Comment by wumpz** [comment]
To somehow document this: https://stackoverflow.com/questions/5813311/no-newline-at-end-of-file.

This weird

```diff
\ No newline at end of file
```

is put into a diff by Git to signal, that there is no linefeed at the end of the last line of this diff.

**Comment by wumpz** [comment]
So my implementation stores this as a flag within **UnifiedDiffFile**, so the write can write those lines without putting a linefeed at the end. I hope that is right.

**Comment by wumpz** [comment]
Just deployed a new version.

**Comment by elanzini** [comment]
@wumpz I am still getting the same exact error when I try to parse [this patch](https://bitbucket.org/asomov/snakeyaml/issues/attachments/377/asomov/snake-git/1582961339.7/377/wiki.patch) from[ this JIRA ticket](https://bitbucket.org/asomov/snakeyaml/issues/377).

**Comment by wumpz** [comment]
@elanzini I will look into it.

**Comment by wumpz** [comment]
@elanzini May I include this patch as test data into java-diff-utils? I already solved the problem and would like to use your test data.

**Comment by elanzini** [comment]
@wumpz by all means! I found other examples, I will add them to #122. And thanks for the quick fix!

**Comment by elanzini** [comment]
@wumpz I found other examples for this problem https://github.com/pgjdbc/pgjdbc/commit/cdeeaca47dc3bc6f727c79a582c9e4123099526e or https://github.com/apache/cxf/commit/1cf4fed546904a4a2560f53a2a2391d834b4026. For both, if you try to parse the diff, you get the same complaint `WARNING:   >>> no rule matched \ No newline at end of file`.

**Comment by wumpz** [comment]
@elanzini I included all three of your tests. They are now succeeding.

**Comment by williamfzc** [comment]
same problem here :) 
I just found that the latest released version (4.10) did not contain this change
When will 4.11 will be released? Looking forward to it and thanks!

**Comment by wumpz** [comment]
Still no valid date there, since I wanted to include the new implementation of meyers linear space diff algorithm.
